### PR TITLE
[camera_android] Support concurrently image capture and image streaming

### PR DESCRIPTION
**Pull request by camsim99**
Properly configures surface needed for image capture when image streaming/recording is started to support concurrently still capture and image streaming.

Fixes https://github.com/flutter/flutter/issues/125314.

Apologies for the many commits :(

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java [hunk 1]
```diff
@@ -1140,10 +1149,15 @@ public void startPreviewWithImageStream(EventChannel imageStreamChannel)
   public void onImageAvailable(ImageReader reader) {
     Log.i(TAG, "onImageAvailable");
 
+    // Use acquireNextImage since image reader is only for one image.
+    Image image = reader.acquireNextImage();
+    if (image == null) {
+      return;
+    }
+
```
Comment (camsim99) [inline packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java:1157]: Unrelated fix: `acquireNextImage` may return null, so added null check.

File: packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java [hunk 2]
```diff
@@ -659,7 +669,6 @@ public void onCaptureCompleted(
         };
 
     try {
-      captureSession.stopRepeating();
```
Comment (camsim99) [inline packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java:662]: Unrelated fix: This call should not be necessary for still capture to proceed.

**Review by johnmccutchan — APPROVED** [review]

**Review by bparrishMines — APPROVED** [review]
LGTM

I believe we were originally concerned with attempting preview, image streaming, and still image capture simultaneously due to hardware limitations on some devices. But it looks like all hardware supports all three simultaneously: https://developer.android.com/reference/android/hardware/camera2/CameraDevice#regular-capture.

However they can be limited by resolutions.

**Comment by camsim99** [comment]
> I believe we were originally concerned with attempting preview, image streaming, and still image capture simultaneously due to hardware limitations on some devices. But it looks like all hardware supports all three simultaneously:

Oh, I did not realize this, but that context makes sense. Definitely good to be aware of, though, if we start seeing related bugs. Thanks!
